backend/app/models/command.rb

```diff
@@ -3,4 +3,87 @@ class Command < ActiveRecord::Base
   has_many :device_commands, dependent: :destroy
   has_many :devices, through: :device_commands
 
+  serialize :settings, Hash
+
+  def plist_values
+    {RequestType: type.gsub(/Command$/, '')}
+  end
+
+  def parse_response!(response, device)
+  end
+
+end
+
+class DeviceInformationCommand < Command
+
+  GeneralQueries = %w(UDID Languages Locales DeviceID OrganizationInfo LastCloudBackupDate).freeze
+  DeviceInformationQueries = %w(DeviceName OSVersion BuildVersion ModelName Model ProductName SerialNumber DeviceCapacity AvailableDeviceCapacity BatteryLevel CellularTechnology IMEI MEID ModemFirmwareVersion IsSupervised IsDeviceLocatorServiceEnabled IsActivationLockEnabled IsDoNotDisturbInEffect DeviceID EASDeviceIdentifier IsCloudBackupEnabled).freeze
+  NetworkInformationQueries = %w(ICCID BluetoothMAC WiFiMAC EthernetMACs CurrentCarrierNetwork SIMCarrierNetwork SubscriberCarrierNetwork CarrierSettingsVersion PhoneNumber VoiceRoamingEnabled DataRoamingEnabled IsRoaming PersonalHotspotEnabled SubscriberMCC SubscriberMNC CurrentMCC CurrentMNC).freeze
+  ItunesStoreAccountQueries = %w(iTunesStoreAccountIsActive iTunesStoreAccountHash)
+
+  def plist_values
+    super.merge({
+      Queries: GeneralQueries | DeviceInformationQueries | NetworkInformationQueries | ItunesStoreAccountQueries
+    })
+  end
+
+  def parse_response!(response, device)
+    response = response['QueryResponses']
+    device_model = DeviceModel.find_or_create_by(model: response['ProductName'])
+    major, minor, patch = response['OSVersion'].split('.')
+    patch ||= 0
+    device_firmware = DeviceFirmware.find_or_create_by(buildid: response['BuildVersion'], major: major, minor: minor, patch: patch)
+    device_model_firmware = DeviceModelFirmware.find_or_create_by(device_firmware: device_firmware, device_model: device_model)
+    device.update(
```

Comment:
Really don't like persisting all the values like this; should go with a K/V store.